File: convert BST to Greater Tree.cpp
class Solution {
public:
    TreeNode* traversal(TreeNode* root,int &s)
    {
        if(root)
        {
            traversal(root->right,s);
            s+=root->val;
            root->val=s;
            traversal(root->left,s);
        }
        return root;
    }
    TreeNode* convertBST(TreeNode* root) {
        int s=0;
        return traversal(root,s);
        
    }
};

File: convert BST to greater tree.java
class Solution {
    public TreeNode traversal(TreeNode root,int[] s)
    {
        if(root!=null)
        {
            traversal(root.right,s);
            s[0]+=root.val;
            root.val=s[0];
            traversal(root.left,s);
        }
        return root;
    }
    public TreeNode convertBST(TreeNode root) {
        int[] s={0};
        return traversal(root,s);
        
        
    }
}
